### PR TITLE
CORE-64 convolution channel out/in check

### PR DESCRIPTION
**Pull request by cottand**

### CODE REVIEW

**Comment by cottand** [comment]
Check:
conv3

File: src/app/ir/checking/findBadConvChannels.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import GraphNode from '@/app/ir/GraphNode';
+import IrError from '@/app/ir/checking/irError';
+import { Severity } from '@/app/ir/checking/severity';
+
+/**
+ * Finds all nodes in the graph that are right before a Conv node and are Conv
+ * nodes themselves, and checks their channels match.
+ */
+export const findBadConvChannels = (graph: Graph) => {
+  type Conv = Conv1d | Conv2d | Conv3d;
```
Comment (KacperKazan) [inline src/app/ir/checking/findBadConvChannels.ts]: ConvTranspose1d, ConvTranspose2d, ConvTranspose3d can also be shape checked in this way. could add them too?

Comment (cottand) [reply src/app/ir/checking/findBadConvChannels.ts]: but does this check happen between 2 layers A and B where A is Conv2d and B is ConvTranspose3d?
And vice versa

Comment (KacperKazan) [reply src/app/ir/checking/findBadConvChannels.ts]: yeah could do. ConvTranspose is kind of like inverse convolution. The in_channels and out_channels check should still be possible between all of them I think

Comment (cottand) [reply src/app/ir/checking/findBadConvChannels.ts]: _you think_ ahahah sure nws

**Review by KacperKazan — APPROVED** [review]

**Pull request merged**